.github/workflows/test.yml

```
@@ -12,51 +12,227 @@ jobs:
         uses: actions/setup-node@v4.1.0
         with:
           node-version: ${{ matrix.node-version }}
-      - name: Node.js modules cache, repository
+      - name: Node.js modules cache
         uses: actions/cache@v4
-        id: repo-cache
+        id: modules-cache
         with:
-          path: ${{ github.workspace }}/node_modules
-          key: ${{ runner.os }}-${{ matrix.node-version }}-modules-repo-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-${{ matrix.node-version }}-modules-repo
-      - name: Node.js modules cache, backend
+          path: |
+            ~/.cache/Cypress
+            ${{ github.workspace }}/node_modules
+            ${{ github.workspace }}/backend/node_modules
+            ${{ github.workspace }}/frontend/node_modules
+          key: ${{ runner.os }}-${{ matrix.node-version }}-modules-${{ hashFiles('**/package-lock.json') }}
+      - name: Install dependencies
+        if: steps.modules-cache.outputs.cache-hit != 'true'
+        run: npm install
+      - name: Check for uncomitted changes
+        run: git diff --exit-code
+
+  Lint:
+    needs: Setup
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Restore cache
         uses: actions/cache@v4
-        id: backend-cache
         with:
-          path: ${{ github.workspace }}/backend/node_modules
-          key: ${{ runner.os }}-${{ matrix.node-version }}-modules-backend-${{ hashFiles('**/backend/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-${{ matrix.node-version }}-modules-backend
-      - name: Node.js modules cache, frontend
+          path: |
+            ~/.cache/Cypress
+            ${{ github.workspace }}/node_modules
+            ${{ github.workspace }}/backend/node_modules
+            ${{ github.workspace }}/frontend/node_modules
+          key: ${{ runner.os }}-18.x-modules-${{ hashFiles('**/package-lock.json') }}
+      - name: Run linting and formatting checks
+        run: npm run test:fix
```

Should we fix the lint errors directly within this workflow? Previously, it only checked for lint errors without fixing them. 

```suggestion
        run: npm run test:lint
```

---

Good spot! Resolved in latest commit.